cloudinstancecredentials-config

```diff
@@ -0,0 +1,39 @@
+#! /usr/bin/python3
+
+# Copyright 2024 SUSE LLC
+#
+# This file is part of cloudinstancecredentials
+#
+# cloudinstancecredentials is free software: you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cloudinstancecredentials is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# cloudinstancecredentials. If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+from cloudinstancecredentials import (
+    frameworkfactory,
+    config
+)
+
+# PROCESS
+log = logging.getLogger('config-credentials')
+metadata = frameworkfactory.get_metadata(log)
+
+if metadata:
```

Comment:
maybe it can be only a `try/except` block instead of the mix of if + try

```python
    try:
        username = metadata.username()
        password = metadata.password()
        config.ConfigCredentialSetter(log, username, password).set_credentials()
        log.info('Done.')
    except KeyError as err:
        # handle this exception
     except AttributeError (or the proper exception instead of the esle):
         log.warning('No instance metadata. Process finished.')
    ```

Reply:
Updated in fb874e5 to a guard clause.